### DISCUSSION TITLE
Alarm 5: Probe did not contact the workpiece

### DISCUSSION THREAD

**Discussion by johnvolbeda** [winder/Universal-G-Code-Sender#2298] [Help]
I am using the Autoleveler plugin with an X, Y,Z touch probe and get this error shown in the title.  I load the file and select Use Loaded File. The Min and Max values for X, Y and Z auto populated and I do not change these values.  I understand the purpose of the Resolution settings. 

The level probe stops short of the material.  Part of my problem is that I do not know what the settings should be for the following parameters:

Gcode Z surface
SETTINGS TAB:
Z Height of probe surface in gcode
I assume the Probe X, Y and Z offset are zero since I move the touch probe directly over the material and then reset X, Y, and Z.

Any help or guidance is appreciated.

FYI, I am engraving text on a 1 inch diameter acrylic convex disk and need to scan the surface and apply the contour to the Gcode so the engraved depth will be uniform on the disc.

**Comment by breiler** [comment]
There has been a lot of fixes and adjustments to the autoleveler module lately. I'd recommend the latest freshly baked version from here: https://github.com/winder/Universal-G-Code-Sender/releases/tag/v2.0.20

I have tested probing on this version quite thoroughly but there is always a risk that I have missed something. So if it is possible be careful when initiating a probe.

![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/1805f0f5-2382-4793-8c51-069a350c97d0)

When you use the `Use Loaded File` to get the model size it will set the probe area to the exact extent of the model. This usually means that you will never reach material with the probe. Manually set the Z min to be slightly under the surface material. 

The `Z Retract` will now be displayed as percent in the latest version, leave it to 100% to begin with. By lowering this you can speed up the probe process, but if you are not careful it could cause the bit to scratch the surface of the material.

**Reply by johnvolbeda** [reply]
(Looks like my reply from yesterday did not post?)  Thank you so much for your guidance and suggestions.  I am making slow progress but I am having intermittent operation issues completing the AUTOLEVEL routine.  Earlier, I could load the .cng file, VISUALIZE the graphics and adjust the AUTOLEVEL parameters as you said and even probe the part using my level probe.  But I was not able to change the RESOLUTION but it remained as four points on each corner of the .cng graphics file on the VISUALIZER window.

Now, after starting UGS 2.0.20, I loading the .cgn file, I can see the graphics momentarily on the VISUALIZER but then disappears. I have reinstalled UGS 2.0.20 several times but no change is noticed.  

One more update ... when I run this on a PC computer, I can scan the part and the only function that is missing is changing the RESOLUTION setting ... it remains at 0.76 with only six probe points.  I hope this is helpful.

I suspect this is an operator error (me) but not sure how to proceed or what my next steps should be.

UGS 2.0.20
GRBL1.1f
Xpro V3 CNC board
MACOS 10.14.5

**Reply by breiler** [reply]
> Now, after starting UGS 2.0.20, I loading the .cgn file, I can see the graphics momentarily on the VISUALIZER but then disappears. I have reinstalled UGS 2.0.20 several times but no change is noticed.

In version 2.0.20 I changed the behaviour so that the autoleveler preview is only visible when the autoleveler window is showing. I have seen many demonstration videos from people where they have both the probing and autolevel visualization active at the same time even if they weren't using those functions at the moment. Is the visualization not showing even if the autolevel window is showing?

>But I was not able to change the RESOLUTION but it remained as four points on each corner of the .cng graphics file on the VISUALIZER window.

What happens if you manually type "0.1" in the resolution box?
What are your other settings for min/max?

**Reply by johnvolbeda** [reply]
After some testing, the difficulty may be related to the computer type and operating system I am using.  I have tried three computer platforms with UGS 2.0.20 installed on each of them. The MACOS with 10.14.5 and PC with Window 7 does not display correctly as mentioned earlier. But using PC with Windows 10 on a third computer, the graphics display is correct as expected. The vizualizer and the autolever windows are showing correctly.  I can also change the Resolution.  Answers to your questions:  (1) when I enter "0.1" resolution the black dots fill the screen.  (2) The min/max settings (mm): X Min: -9.977, X Max: 9.98,  Y Min: -4.43, Y Max: 4.427, Z Min: 0.0, Z Max: 6.35.  I am upgrading my CNC control board to a newer version and will report back later.  Thank you so much for taking time to reply.

**Reply by johnvolbeda** [reply]
Update: I am using a dedicated PC with WIN 10 and replaced the CNC Controller board (Xpro V5) and I can now load UGS 2.0.20 and the .ngs file.  I am able to use the AUTOLEVEL plugin and see the VISUALIZER correctly.

Using the small part per the previous email, I changed Z Min from 0 to -0.25, Z Max from 2.0 to 0.2, Z Retract was changed from 100% to 90%, Z height of probe surface in gcode=0, resolution = 0.26 with about 15 probe points.  When scanning the part, I get the following error code and the level probe tip stop above the part that is being scanned:

\ The communicator has been paused ****
An error was detected while sending 'G38.2 Z-0.45 F0.394': (ALARM:4) Probe fail. Probe is not in the expected initial state before starting probe cycle when G38.2 and G38.3 is not triggered and G38.4 and G38.5 is triggered. Streaming has been paused.

Your help and directions are appreciated sincerely.

**Reply by breiler** [reply]
The error says that it couldn't find the surface at the given distance. Sound like your Z Min is still set too high. Have you tried setting it to -0.50?

**Reply by johnvolbeda** [reply]
Thanks so much for your quick reply.  Changing Z Min to -0.5 gives the following error message:
An error was detected while sending 'G38.2 Z-0.7 F0.394': (ALARM:4) Probe fail. Probe is not in the expected initial state before starting probe cycle when G38.2 and G38.3 is not triggered and G38.4 and G38.5 is triggered. Streaming has been paused.

**Reply by breiler** [reply]
And this sounds like the probe was not cleared from the material when it started the probe sequence.

Without seeing how the probing is actually behaving or the material that you are probing I am guessing that you need to change these settings:
* Set the Z retract to 100% - This will make sure that the probe is lifted to the max height
* Increase the Z max to 0.5 - This should be as high as the highest point of the material

**Reply by johnvolbeda** [reply]
Thank you for the review and reply.  The probe starting position is just above the material.  When I change Z retract to 100% and Z max to 0.5, I get the following error:  An error was detected while sending 'G38.2 Z-1 F0.394': (ALARM:4) Probe fail. Probe is not in the expected initial state before starting probe cycle when G38.2 and G38.3 is not triggered and G38.4 and G38.5 is triggered. Streaming has been paused.
ok

I attached a screen shot and hopefully you can see it and the end of the run with the error.

I value your time and expertise. 

![Screen Shot](https://github.com/winder/Universal-G-Code-Sender/assets/41496797/60e67488-ce77-4f7b-98d6-fd6a38b48131)

**Reply by breiler** [reply]
OK, I see the likely cause for the problem. The gcode is in inches and you are using millimeters in the UI. The probe settings should therefore also be in millimeters. But something is clearly not working as your probe settings are intepreted and vizualised as inches.

I am almost certain that I have tested this scenario, but apparently I have missed something.

Until I find a fix for this problem, is it possible to regenerate the gcode in millimeters?

**Reply by breiler** [reply]
Or set the units to inches in the UI if that is your preferred units:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/e0b87a00-d8b9-4a19-bb46-0cdb3a32ede9)

**Reply by johnvolbeda** [reply]
Thank you so much for noticing.  Of course, I will try both to (1) change the units to mm to generate the gcode and later on (2) change the units to inches in UI.  Please stand by ....

**Reply by johnvolbeda** [reply]
Good morning.   I tested various options: (1) change the units of the gcode to mm and set the UI to mm.  (2) Change the units in the UI and used the gcode generated in inches.  I changed to values of Z Min, Z Max, Z Retract to various settings and all of them result in the probe not touching the surface of the material.  The error code is: An error was detected while sending 'G38.2 Z-7.1 F10': (ALARM:4) Probe fail. Probe is not in the expected initial state before starting probe cycle when G38.2 and G38.3 is not triggered and G38.4 and G38.5 is triggered. Streaming has been paused. 

I did notice than when the Z Max value is lowered to .01, it brings the end of the probe closer to the material, but still not enough to make contact with the material.

I believe we are eliminating variable and am sure we are getting closer.  Let me know what other information you would like.

I inserted a screen shot in the next block.

Thank you in advance.

**Comment by johnvolbeda** [comment]
![Screen Shot 13 Sep 2023](https://github.com/winder/Universal-G-Code-Sender/assets/41496797/1b68b349-f329-42d3-81e2-26cedb7212ba)

**Reply by breiler** [reply]
From this log it looks like it just sent the command for the first probe command and complains that the probe was already triggered from start.

Was "ALARM: PROBE" active before you started probing?
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/8a9716e9-da1e-454e-a8b9-c18c01d087c5)

**Reply by johnvolbeda** [reply]
Thank you for your quick reply.  It is probably my ignorance on the operation of many of these features.  But yes, the ALARM: PROBE is energized on start up.  I also checked into the GRBL documentation and says the following:

$6 - Probe pin invert, boolean
By default, the probe pin is held normally-high with the Arduino's internal pull-up resistor. When the probe pin is low, Grbl interprets this as triggered. For the opposite behavior, just invert the probe pin by typing $6=1. Disable with $6=0. You may need a power cycle to load the change.

$6=0 but when I changed it to $6=1, the ALARM: PROBE is de-energized.  After adjusting some of the AUTO LEVELER settings,  I can now for the **first time** probe the convex surface!!!  I will insert a screen shot of the finished part.  Tomorrow, I will do some more testing but I think that with your great help and patience, I just might be able to move forward and engrave on a convex surface.  More soon and then we can close this out!

**Comment by johnvolbeda** [comment]
![Convex Surface Test 2 13 Sep 2023](https://github.com/winder/Universal-G-Code-Sender/assets/41496797/86fd16c6-58e8-42a8-be5f-13ee1103e3ef)

**Reply by breiler** [reply]
Cool!

There are some obvious fixes that would have improved the user experience, for instance:
* You should not be able to start probing if the probe is already active
* Display the units of the settings
* And have a look at the bug that didn't properly update the settings

**Reply by johnvolbeda** [reply]
After some more testing, it appears mixing of inches and mm units between the Gcode generated by F-Engrave and UGS UI could be confusing to the operation. I had success when the Gcode was generated using mm and setting the UI to mm.  After scanning the part, applying the result to the Gcode, installing the router bit and setting zero in the center of the part and sending the code, the starting point of the tool path is at the bottom left side of the letter "O" in my text.  However, the origin in F-Engrave is mid center.  Since this is a convex shaped part, the bit does not touch the part.  This is confusing to me and not sure how to correct this.  (I did not close this discussion since this may still be related to the AUTOLEVEL feature.)  Your expertise is greatly appreciated.

**Comment by johnvolbeda** [comment]
My apologies for not providing an update on my issues.  This has totally been resolved and the AUTOLEVEL feature indeed works wonderfully!  Thank you so much for all your efforts and keeping this great software available.  We appreciate you!

**Comment by johnvolbeda** [comment]
This has been resolved and is closed!
